src/puppetlabs/trapperkeeper/services/webserver/jetty9_service.clj

```diff
@@ -26,23 +33,38 @@
 (defservice jetty9-service
   "Provides a Jetty 9 web server as a service"
   WebserverService
-  [[:ConfigService get-in-config]]
+  {:required [ConfigService]
+   :optional [FilesystemWatchService]}
   (init [this context]
         (log/info (i18n/trs "Initializing web server(s)."))
-        (let [config (or (get-in-config [:webserver])
+        (let [config-service (get-service this :ConfigService)
+              config (or (tk-config/get-in-config config-service [:webserver])
                          ;; Here for backward compatibility with existing projects
-                         (get-in-config [:jetty])
+                         (tk-config/get-in-config config-service [:jetty])
                          {})]
           (config/validate-config config)
           (core/init! context config)))
 
   (start [this context]
          (log/info (i18n/trs "Starting web server(s)."))
-         (let [config (or (get-in-config [:webserver])
+         (let [config-service (get-service this :ConfigService)
+               config (or (tk-config/get-in-config config-service [:webserver])
                           ;; Here for backward compatibility with existing projects
-                          (get-in-config [:jetty])
-                          {})]
-           (core/start! context config)))
+                          (tk-config/get-in-config config-service [:jetty])
+                          {})
+               started-context (core/start! context config)]
+           (if-let [filesystem-watcher-service
+                    (maybe-get-service this :FilesystemWatchService)]
+             (let [watcher (watch-protocol/create-watcher filesystem-watcher-service)]
+               (doseq [server (:jetty9-servers started-context)]
+                 (when-let [ssl-context-factory (-> server
+                                                    second
+                                                    :state
+                                                    deref
+                                                    :ssl-context-factory)]
```

Comment:
I'll try to walk through it. First, `:jetty9-servers` is a map of id, server kv pairs. These pairs can be accessed using vec access calls such as `second`. So the first bit is passing the `server` kv pair to `second`, which returns the server (and drops the id). Inside the server context:

```(def ServerContext
  {:state     (schema/atom ServerContextState)
   :handlers  ContextHandlerCollection
   :server    (schema/maybe Server)})
```

In clojure, using a keyword as a function does a lookup in the map on that keyword, so calling `:state` on the server-context is the equivalent of doing `(get server-context :state)`. Looking at the schema for `:state`, it is an atom holding a `ServerContextState`, so when `deref` is called on the `:state` bit, a ServerContextState is returned. `(deref foo)` is the same as `@foo`.

```(def ServerContextState
  {:mbean-container (schema/maybe MBeanContainer)
   :overrides-read-by-webserver schema/Bool
   :overrides (schema/maybe {schema/Keyword schema/Any})
   :endpoints RegisteredEndpoints
   :ssl-context-factory (schema/maybe InternalSslContextFactory)})
```

The last item in the threading macro is another map lookup, which itself returns an `InternalSslContextFactory`. 
How was that?